### PR TITLE
bug fix referer header existence not being checked

### LINKED ISSUE #10859: [beta] 500 TypeError: Cannot read property 'indexOf' of undefined
Steps to reproduce:
- You need to be logged in
- Open new tab / window in your browser
- Visit homepage

You will see: `500 TypeError: Cannot read property 'indexOf' of undefined`

See snippets below:

```
// server/boot/home.js
// ...
module.exports = function(app) {
  // ...
  function index(req, res, next) {
    // ...
    const referer = req.headers.referer;
  
    if (req.user) {
      if ((referer.indexOf('/settings') >= 1) && (req.user.isGithubCool)) {
        // ...
      }
      // ...
    }
    // ...
  }
  // ...
}

```

`referer` header is not always exist so it's better to add conditional to check that

### PR DESCRIPTION
**Pull request by ivantedja**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #10859

### CODE REVIEW

File: server/boot/home.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports = function(app) {
     const referer = req.headers.referer;
 
     if (req.user) {
-      if ((referer.indexOf('/settings') >= 1) && (req.user.isGithubCool)) {
+      if (referer && (referer.indexOf('/settings') >= 1) && (req.user.isGithubCool)) {
```
Comment (Bouncey) [inline server/boot/home.js]: Needs a `\n` in here to pass Travis-CI 80char limit

File: server/boot/home.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ module.exports = function(app) {
     const referer = req.headers.referer;
```
Comment (BerkeleyTrue) [inline server/boot/home.js]: To make it cleaner you could use destructuring with default params

```js
const { referer = '/'  } = req.headers;
```

**Comment by dhcodes** [comment]
@ivantedja looks like the travis build is failing due to line length being >80 characters. You can fix that up and amend the commit to get the travis build to pass.

File: server/boot/home.js (hunk 2 above)
Comment (ivantedja) [inline server/boot/home.js]: that's neat! I'll initialize it with `''` instead of `'/'` because the value is not exist

**Comment by ivantedja** [comment]
@Bouncey @dhcodes thanks, I was not aware of 80 chars limit 😸

**Comment by dhcodes** [comment]
This fixed my local issue. I think it's good to go.

**Comment by BerkeleyTrue** [comment]
LGTM

**Comment by dhcodes** [comment]
@ivantedja maybe open a separate issue regarding the linting question.

**Pull request merged**